### PR TITLE
feat(stdlib): use table.fill() when aggregateWindow(createEmpty: true) is used

### LINKED ISSUE #3680: Add fill empty transformation to aggregate window
The fill empty transformation described [here](https://github.com/influxdata/flux/issues/3679) should be added to aggregate window. It should go right after the aggregate call but before the duplicate or second window.

I think roughly like this:

```
aggregateWindow = (every, fn, offset=0s, column="_value", timeSrc="_stop", timeDst="_time", createEmpty=true, tables=<-) => tables
    |> window(every: every, offset: offset, createEmpty: createEmpty)
    |> fn(column: column)
    |> ((tables=<-) => if createEmpty then tables |> _fillEmpty() else tables)()
    |> duplicate(column: timeSrc, as: timeDst)
    |> window(every: inf, timeColumn: timeDst)
```

DOD:
* An acceptance test for this functionality.
* The acceptance test should be extended to test the push down from https://github.com/influxdata/influxdb/pull/21519.

### PR DESCRIPTION
**Pull request by jsternberg**
Fixes #3680.

### Done checklist
- [x] docs/SPEC.md updated
- [x] Test cases written

### CODE REVIEW

**Review by scbrickley — COMMENTED** [review]
I had some questions about the new testcases. Hoping you can clarify things for me a bit before I hit approve.

File: stdlib/universe/aggregate_window_mean_test.flux
Comment (jsternberg) [reply stdlib/universe/aggregate_window_mean_test.flux:35]: This was me just moving the old test case. I wanted the name `aggregate_window_test` for the new system which allows multiple test cases in a single file so I moved this one to keep the old test.

File: stdlib/universe/aggregate_window_test.flux
Comment (jsternberg) [reply stdlib/universe/aggregate_window_test.flux:81]: This fill is meant to ensure that empty tables have at least one null row so that we can use `fill()` to fill the values. This is testing the new `table.fill()` and not the one in universe.

File: stdlib/universe/aggregate_window_test.flux
Comment (jsternberg) [reply stdlib/universe/aggregate_window_test.flux:21]: The output is intended to have nulls. Maybe a better name is `with_empty`?

File: stdlib/universe/aggregate_window_test.flux
Comment (scbrickley) [reply stdlib/universe/aggregate_window_test.flux:81]: Ah, I think I get it now. We're inserting null values at specific timestamps where we need a value but don't have one.

Thanks for clarifying!

File: stdlib/universe/aggregate_window_test.flux
Comment (scbrickley) [reply stdlib/universe/aggregate_window_test.flux:21]: I think the name is fine now that I have a better understanding of what the new behavior is. I'll leave that up to you.

**Comment by jsternberg** [comment]
You can approve. It seems there are some test cases that are failing. The primary reason this is a draft PR is to prevent it from being merged since some influxdb work needs to be done first.

**Review by scbrickley — APPROVED** [review]

**Comment by jsternberg** [comment]
I've left this as a draft so it doesn't get accidentally merged yet, but the PR is complete now so I requested a re-review. I had to change existing test cases for the new behavior so please pay attention to make sure those changes make sense. Thanks.

File: etc/checkfmt.sh
Comment (scbrickley) [inline etc/checkfmt.sh:41]: Thanks for catching this :+1:

File: interpreter/interpreter_test.go
Comment (scbrickley) [inline interpreter/interpreter_test.go:867]: :+1:

File: stdlib/universe/universe.flux
Comment (scbrickley) [inline stdlib/universe/universe.flux:214]: This is tangential to your changes, but I just wanted to comment on the formatting here, which looks odd to me. I think I would prefer something like:

```suggestion
_fillEmpty = (tables=<-, createEmpty) => if createEmpty then
    tables |> table.fill()
else
    tables
```

But obviously that's not your responsibility for this PR. Just wanted to make note of it in case we wanted to file an issue for it later.

File: stdlib/planner/group_max_test.flux
Comment (scbrickley) [inline stdlib/planner/group_max_test.flux:150]: I'll leave this to your judgment, but I think it would be helpful to see a test case with multiple rows in the `want` output for when when we set `createEmpty` to `false`. Something like what you have in the `group_max_agg_window_empty` testcase, but omitting all the would-be empty rows. Thoughts?

**Review by scbrickley — APPROVED** [review]
LGTM - I had one suggestion (feel free to take it or leave it), and one small formatting nitpick unrelated to your work here.

**Comment by jsternberg** [comment]
A potential complication to merging this. `createEmpty` defaults to true, but it acted like `createEmpty: false` was always and irreversibly on whenever a selector was used. So it is possible that somebody wrote an `aggregateWindow` with a selector, got the behavior of `createEmpty: false`, but never explicitly specified it. These queries would potentially be breaking changes when this gets merged as the output for them would change.

**Pull request merged**
